File: category/bfs/PGS_43162_네트워크.js
//네트워크 bfs풀이
function solution(n, computers){
    const visited = Array(n).fill(false);
    let answer = 0;

    const bfs = (start) => {
        const queue = [start];
        visited[start] = true;
        while(queue.length > 0){
            const v = queue.shift();
            
            for (let u = 0; u < n; u++){
                if (computers[u][v] === 1 && !visited[u]){
                    visited[u] = true;
                    queue.push(u);
                }
            }
        }
    }

    for (let i = 0; i < n; i++){
        if (visited[i] === false){
            bfs(i);
            answer ++;
        }
    }

    return answer;

}

File: category/bfs/PGS_43162_네트워크.py
#동일 문제 BFS 접근법
from collections import deque

def solution(n,computers):
    answer = 0
    visited = [False] * n

    def bfs(start):
        q = deque([start])
        visited[start] = True
        while q:
            v = q.popleft()
            if visited[v] == True:
                continue
            visited[v] = True
            for u in range(n):
                if computers[v][u] == 1 and not visited[v]:
                    visited[u] = True
                    q.push(u)


    for i in range(n):
        if not visited[i]:
            bfs(i)
            answer += 1

    return answer